contrib_bots/lib/giphy.py

```diff
@@ -0,0 +1,106 @@
+# To use this plugin, you need to set up the Giphy API key for this bot in
+# zulip_giphy_config.py
+
+from __future__ import absolute_import
+import requests
+from . import zulip_giphy_config as config
+
+
+class GiphyHandler(object):
+    '''
+    This plugin facilitates giving a gif image that appropriate with the
+    keywords user input. The gif image and search algorithm are provided by
+    giphy. It looks for messages starting with "@giphy" or the mention that led
+    to the bot, then we send a gif image that appropriate with keywords. You
+    also can send the keyword via private message with bot.
+    '''
```

Comment:
I'm wondering about phrasing here, mainly how to make it clear how the bot actually works:

- Which image from the search is posted for a given keyword? While testing the same keyword repeatedly I've noticed that different GIFs are posted for the same keyword, but sometimes they are repeated. 

- I think we should use capitalized Giphy everywhere since that's the name of the company.

- I'm going to suggest a direct rephrasing since I know what you've meant and going back and forth on grammar doesn't make sense with this task :) 

```
This plugin posts a GIF in response to the keywords provided by the user.
Images are provided by Giphy, through the public API.
The bot looks for messages starting with "@giphy" or @mention of the bot
and responds with a message with the GIF based on provided keywords.
It also responds to private messages.
```

- How many keywords can a user provide?